### PR TITLE
feat: Add members tab to project settings

### PR DESCRIPTION
**Pull request by mgs95**

### CODE REVIEW

File: kdl-client/src/Pages/Server/pages/Server/pages/Project/Project.tsx [hunk 1]
```diff
@@ -78,6 +89,23 @@ function Project() {
...
           <UpdateRepository project={project} close={hideRepoEdit} />
         </Panel>
+        <Panel
+          title="Member details"
+          show={memberDetails !== null}
+          close={() => setMemberDetails(null)}
+          noShrink
+          dark
+        >
+          {memberDetails === null ? (
```
Comment (marco-gattelli) [inline kdl-client/src/Pages/Server/pages/Server/pages/Project/Project.tsx:99]: Are you planning to add something when `memberDetails` is `null`? If so then ok but if not we can refactor this:
```
{memberDetails && (<MemberDetails
              member={memberDetails}
              close={() => setMemberDetails(null)}
              projectId={project.id}
            />)}
```

File: kdl-client/src/Pages/Server/pages/Server/pages/Project/components/MemberDetails/MemberDetails.tsx [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+type Props = {
+  member: GetProjectMembers_project_members;
+  projectId: string;
+  close: () => void;
+};
+function MemberDetail({ member, projectId, close }: Props) {
+  const [updateMember] = useMutation<UpdateMember, UpdateMemberVariables>(
+    UpdateMemberMutation,
+    {
+      onError: (e) => console.error(`updateMember: ${e}`),
```
Comment (marco-gattelli) [inline kdl-client/src/Pages/Server/pages/Server/pages/Project/components/MemberDetails/MemberDetails.tsx:51]: Shouldn't you update the cache when a member has been updated?

File: kdl-client/src/Pages/Server/pages/Server/pages/Project/components/TabMembers/TabMembers.tsx [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+
+          cache.writeQuery({
+            query: GetMembersQuery,
+            variables: {
+              id: projectId,
+            },
+            data: {
+              project: {
+                ...project,
+                members: project.members.concat(newMembers),
```
Comment (marco-gattelli) [inline kdl-client/src/Pages/Server/pages/Server/pages/Project/components/TabMembers/TabMembers.tsx:65]: Can't we use the spread operator?
`members: [...project.members, ...newMembers]`

File: kdl-client/src/Pages/Server/pages/Server/pages/Project/components/TabMembers/TabMembers.tsx [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+    },
+  });
+
+  // Update opened member details as data is updated
+  useEffect(() => {
+    if (dataMembers && memberDetails) {
+      const selectedMembers = dataMembers.project.members.filter(
+        (m) => m.id === memberDetails.id
+      );
+      const selectedMember = selectedMembers[0];
```
Comment (marco-gattelli) [inline kdl-client/src/Pages/Server/pages/Server/pages/Project/components/TabMembers/TabMembers.tsx:95]: why do you use this const? is it for for better understanding?

File: kdl-client/src/Pages/Server/pages/Server/pages/Project/components/TabMembers/TabMembers.tsx [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+  }, [dataMembers, openMemberDetails, memberDetails]);
+
+  if (loadingMembers || loadingUsers) return <SpinnerCircular />;
+  if (!dataMembers || !dataUsers || errorMembers || errorUsers)
+    return <ErrorMessage />;
+
+  const users = dataUsers.users.map((user) => user.email);
+  const members = dataMembers.project.members.map((member) => member.email);
+  const options = users.filter(
+    (email) => ![...members, ...memberSelection].includes(email)
```
Comment (marco-gattelli) [inline kdl-client/src/Pages/Server/pages/Server/pages/Project/components/TabMembers/TabMembers.tsx:107]: for each filter iteration you create a new array using the spread operator, we can create a const and use it in the filter:
```
const allMembers = [...members, ...memberSelection]
const options = users.filter((email) => !allMembers.includes(email)
```

File: kdl-client/src/Pages/Server/pages/Server/pages/Project/components/MemberDetails/MemberDetails.tsx [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+          });
+        }
+      },
+    }
+  );
+
+  const { handleSubmit, setValue, unregister, register, watch } = useForm<
+    FormData
+  >({
+    defaultValues: { accessLevel: member.accessLevel },
```
Comment (marco-gattelli) [inline kdl-client/src/Pages/Server/pages/Server/pages/Project/components/MemberDetails/MemberDetails.tsx:93]: Why are you using the `useForm` when you have just one field? I am working on something similar, just a text input and I was wondering if I have to use the `useForm` hook or with a simple `useState` is ok :)

File: kdl-client/src/Pages/Server/pages/Server/pages/Project/components/MemberDetails/MemberDetails.tsx (hunk 2 above)
Comment (mgs95) [reply kdl-client/src/Pages/Server/pages/Server/pages/Project/components/MemberDetails/MemberDetails.tsx:51]: No, updates are automatically updated on cache (exceptios happen when you are adding or removing something)

File: kdl-client/src/Pages/Server/pages/Server/pages/Project/components/TabMembers/TabMembers.tsx (hunk 3 above)
Comment (mgs95) [reply kdl-client/src/Pages/Server/pages/Server/pages/Project/components/TabMembers/TabMembers.tsx:65]: done

File: kdl-client/src/Pages/Server/pages/Server/pages/Project/components/TabMembers/TabMembers.tsx (hunk 4 above)
Comment (mgs95) [reply kdl-client/src/Pages/Server/pages/Server/pages/Project/components/TabMembers/TabMembers.tsx:95]: done

File: kdl-client/src/Pages/Server/pages/Server/pages/Project/components/TabMembers/TabMembers.tsx (hunk 5 above)
Comment (mgs95) [reply kdl-client/src/Pages/Server/pages/Server/pages/Project/components/TabMembers/TabMembers.tsx:107]: done

File: kdl-client/src/Pages/Server/pages/Server/pages/Project/components/MemberDetails/MemberDetails.tsx (hunk 6 above)
Comment (mgs95) [reply kdl-client/src/Pages/Server/pages/Server/pages/Project/components/MemberDetails/MemberDetails.tsx:93]: better useForm

**Pull request merged**